common/src/main/java/com/wynntils/models/lootrun/LootrunModel.java

```diff
@@ -307,11 +309,26 @@ public void onEntitySpawn(SetEntityDataEvent event) {
                 if (!(packedItem.value() instanceof ItemStack itemStack)) return;
 
                 Optional<GearItem> gearItemOpt = Models.Item.asWynnItem(itemStack, GearItem.class);
-                if (gearItemOpt.isEmpty()) return;
+                if (gearItemOpt.isPresent()) {
+                    GearItem gearItem = gearItemOpt.get();
+
+                    if (gearItem.getGearTier() == GearTier.MYTHIC) {
+                        WynntilsMod.postEvent(new MythicFoundEvent(itemStack, true));
+                        dryPulls.store(0);
+                    }
+                    return;
+                }
 
-                GearItem gearItem = gearItemOpt.get();
+                // No need to check tier for these as they are only mythic
+                Optional<InsulatorItem> insulatorItemOpt = Models.Item.asWynnItem(itemStack, InsulatorItem.class);
+                if (insulatorItemOpt.isPresent()) {
+                    WynntilsMod.postEvent(new MythicFoundEvent(itemStack, true));
+                    dryPulls.store(0);
+                    return;
+                }
 
-                if (gearItem.getGearTier() == GearTier.MYTHIC) {
+                Optional<SimulatorItem> simulatorItemOpt = Models.Item.asWynnItem(itemStack, SimulatorItem.class);
+                if (simulatorItemOpt.isPresent()) {
                     WynntilsMod.postEvent(new MythicFoundEvent(itemStack, true));
                     dryPulls.store(0);
                 }
```

Comment:
Repeat yourself twice, but the third time :). Can you add a "mythicFound" boolean and do the remaining work there? Maybe do this in the feature too.